app/models/maintenance_tasks/run.rb

```diff
@@ -499,5 +505,30 @@ def argument_filter
         Rails.application.config.filter_parameters + task.masked_arguments,
       )
     end
+
+    def with_stale_object_retry(max_retries = MAX_STALE_OBJECT_RETRIES)
+      attempt = 1
+      begin
```

Comment:
Style issue but I think we can get rid of the begin by defining `attempt` as a method parameter:
```suggestion
    def with_stale_object_retry(attempt = 1, max_retries = MAX_STALE_OBJECT_RETRIES)
```